Editor: Move editor-sidebar CSS reset closer to its target

alshakero opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
The editor-sidebar children (such as `editor-post-panel__row-label`) need `box-sizing: inherit (border-box)` to look visually aligned. In most environments they get this CSS from the [CSS reset](https://github.com/WordPress/gutenberg/blob/74c71e23c8b974d0b3ece0d22ff4b6b86df45b4f/packages/edit-post/src/style.scss#L22-L25) of the `edit-post` package which is often used to render the editor, and all seems fine. But when the editor is rendered directly from `@wordpress/editor`, i.e without `edit-post` package, this reset is missing, and the sidebar items look misaligned as pictured.

<img width="284" height="676" alt="image" src="https://github.com/user-attachments/assets/effe1a43-2982-4401-a498-f985df024d86" />

## Why?
This moves the CSS reset to where it actually applies. Removing the need for the assumption that the editor will always run inside the `@wordpress/edit-post` package.

## How?
But moving the reset from `edit-post` to the editor itself.

## Testing Instructions
1. Go to [this playground URL](https://playground.wordpress.net/?gutenberg-pr=71096#%7B%22landingPage%22:%22/wp-admin/post-new.php%22,%22features%22:%7B%22networking%22:true%7D,%22steps%22:%5B%7B%22step%22:%22login%22,%22username%22:%22admin%22,%22password%22:%22password%22%7D%5D%7D).
3. Inspect the a sidebar label such as `Status`, try to find `editor-post-panel__row-label`.
4. It should have `box-sizing: inherit;`.
5. The CSS should come from `wp-content/plugins/gutenberg/build/editor/style.css?ver=21.3.0`.
6. Now to compare, you can check [this URL](https://playground.wordpress.net/#%7B%22landingPage%22:%22/wp-admin/post-new.php%22,%22features%22:%7B%22networking%22:true%7D,%22steps%22:%5B%7B%22step%22:%22login%22,%22username%22:%22admin%22,%22password%22:%22password%22%7D%5D%7D) that uses `trunk`, 
7. Inspect the same element, you'll see the CSS comes from `gutenberg/build/edit-post/style.css?ver=21.3.0`.

### Testing Instructions for Keyboard
Unrelated

## Screenshots or screencast <!-- if applicable -->
N/A

alshakero committed:
Move editor-sitebar reset closer to its target

youknowriad commented on packages/editor/src/components/sidebar/style.scss:
```diff
@@ -14,3 +14,7 @@
 	display: none;
 }
+
+.editor-sidebar {
```
Feels like we should be doing the same for all the other `editor-*` classnames that are now in the "edit-post" stylesheet no?

alshakero replied:
Absolutely. I wanted to keep the PR as small as possible to drive the point. I'll move the others.

alshakero committed:
Disperse `edit-post` styles to their original owners

alshakero committed:
Remove editor-sidebar styles from edit-site

alshakero commented on packages/edit-site/src/components/global-styles-sidebar/style.scss:
```diff
@@ -1,22 +1,3 @@
-.editor-sidebar {
-	width: $sidebar-width;
-
-	> .components-panel {
-		border-left: 0;
-		border-right: 0;
-		margin-bottom: -1px;
-		margin-top: -1px;
-
-		> .components-panel__header {
-			background: $gray-100;
-		}
-	}
-
-	.block-editor-block-inspector__card {
-		margin: 0;
-	}
-}
```
I checked these styles one by one. None of them are needed:

1. `width: $sidebar-width;` sets the same width 280px.
2. `border-left: 0; border-right: 0;` are not needed because `.components-panel` has `border: none` now. 
3. `margin-bottom: -1px; margin-top: -1px;` are most likely a trick to collapse the double borders. But there is no border now because of the aforementioned `border: none`. 
4. `background: $gray-100;` This one does change the background from the original `#fff`, to `#f0f0f0`, which I found a source of inconsistency.
5. `block-editor-block-inspector__card`. There is no such element anymore.

alshakero commented:
@youknowriad I moved all the styles to their rightful owners. I tried to give the same treatment to `edit-site` package (the other consumer of `@wordpress/editor`) and I found one instance that [I just removed](https://github.com/WordPress/gutenberg/pull/71096#discussion_r2257649376).
